geoh5py/shared/merging/drape_model.py

```diff
@@ -0,0 +1,171 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from typing import cast
+
+import numpy as np
+
+from ...data import NumericData
+from ...objects import DrapeModel, ObjectBase
+from ...workspace import Workspace
+from .base import BaseMerger
+
+
+class DrapeModelMerger(BaseMerger):
+    _type: type = DrapeModel
+
+    @classmethod
+    def create_object(
+        cls, workspace: Workspace, input_entities: list[ObjectBase], **kwargs
+    ) -> DrapeModel:
+        """
+        Create a new object of type cls._type from a list of input entities.
+        It merges the cells together and create a new object with the merged cells.
+
+        :param workspace: The workspace to create the object in.
+        :param input_entities: The list of input entities to merge together.
+        :param kwargs: The kwargs to pass to the object creation.
+
+        :return: The newly created object merged from input_entities.
+        """
+
+        layers: list = []
+        prisms: list = []
+        previous_prism: int = 0
+        previous_layer: int = 0
+        ghost_prism: np.ndarray = np.array([])
+        ghost_layer: np.ndarray = np.array([])
+
+        for input_entity in input_entities:
+            # get the values of the entity
+            temp_prisms: np.ndarray = cast(
+                np.ndarray, cast(DrapeModel, input_entity).prisms
+            )
+            temp_layers: np.ndarray = cast(
+                np.ndarray, cast(DrapeModel, input_entity).layers
+            )
```

Comment:
```suggestion
            temp_prisms: np.ndarray = input_entity.prisms
            temp_layers: np.ndarray = input_entity.layers
```

Reply:
I agree, but can I shut pylint when it acts stupid?